varcode/coding_effect.py

```diff
@@ -81,33 +53,53 @@ def infer_coding_effect(
     alt : str
         Alternate nucleotides we're replacing the reference with
 
-    cds_offset : int
-        Offset into the coding sequence of the ref->alt substitution
+    transcript_offset : int
+        Offset into the full transcript sequence of the ref->alt substitution
 
     transcript : Transcript
 
     variant : Variant
     """
-    # Don't need a pyfaidx.Sequence object here, just convert it to the an str
-    cds_seq = str(transcript.coding_sequence)
+    if not transcript.complete:
+        raise ValueError(
+            ("Can't annotate coding effect for %s"
+             " on incomplete transcript %s" % (variant, transcript)))
 
-    assert cds_offset < len(cds_seq), \
-        "Expected CDS offset (%d) < |CDS| (%d) for %s on %s" % (
-            cds_offset, len(cds_seq), variant, transcript)
+    sequence = str(transcript.sequence)
+
+    # reference nucleotides found on the transcript, if these don't match
+    # what we were told to expect from the variant then raise an exception
+    transcript_ref = sequence[transcript_offset:transcript_offset + len(ref)]
 
-    # past this point we know that we're somewhere in the coding sequence
-    cds_ref = cds_seq[cds_offset:cds_offset + len(ref)]
     # Make sure that the reference sequence agrees with what we expected
     # from the VCF
-    # TODO: check that the ref allele is correct for UTR by looking
-    # at transcript.sequence instead of transcript.coding_sequence
-    assert cds_ref == ref, \
-        "%s: expected ref '%s' at offset %d of %s, CDS has '%s'" % (
+    assert transcript_ref == ref, \
+        "%s: expected ref '%s' at offset %d of %s, transcript has '%s'" % (
             variant,
             ref,
-            cds_offset,
+            transcript_offset,
             transcript,
-            cds_ref)
+            transcript_ref)
+
+    cds_start_offset = min(transcript.start_codon_spliced_offsets)
+    cds_stop_offset = max(transcript.stop_codon_spliced_offsets)
```

Comment:
All the offsets on `Transcript` (like `start_codon-spliced_offsets`) are relative to the direction off the transcript regardless of which strand it's on.